src/clue/adafruit_clue.py

```diff
@@ -334,13 +721,23 @@ def simple_text_display(
         )
 
     def update_state(self, new_state):
-        self.__update_buttons(new_state)
+        for event in new_state.keys():
+            if event in CONSTANTS.EXPECTED_INPUT_BUTTONS:
+                self.__update_button(event, new_state.get(event))
+            elif event in CONSTANTS.ALL_EXPECTED_INPUT_EVENTS:
+                if self.__state[event] != new_state[event]:
+                    self.__state[event] = new_state.get(event)
 
     # helpers
-    def __update_buttons(self, new_state):
-        # get button pushes
-        for button_name in CONSTANTS.EXPECTED_INPUT_BUTTONS:
-            self.__update_button(button_name, new_state.get(button_name))
+    def __update_button(self, button, value):
+        if button == CONSTANTS.CLUE_STATE.BUTTON_A:
+            if value:
+                self.__state[CONSTANTS.CLUE_STATE.PRESSED_BUTTONS].add("A")
+            self.__state[CONSTANTS.CLUE_STATE.BUTTON_A] = value
+        elif button == CONSTANTS.CLUE_STATE.BUTTON_B:
+            if value:
+                self.__state[CONSTANTS.CLUE_STATE.PRESSED_BUTTONS].add("B")
+            self.__state[CONSTANTS.CLUE_STATE.BUTTON_B] = value

```

Comment:
You can streamline this function by having something like:
`self.__state[button] = value`
at the start (or end) and cleaning up the rest of the conditionals.

Reply:
Updated!